### ISSUE TITLE
[Bug] Optuna sweeper silently changes None type to str.

### ISSUE THREAD

**Issue by YannDubs** [facebookresearch/hydra#1986]
# 🐛 Bug
## Description
When using optuna sweeper if `null` is swept over it will (correctly) be converted to `None` but if it is not swept over it will be converted to the **string** `"None"`. 

## To reproduce

**Minimal Code/Config snippet to reproduce**

examples/my_app.py:

```python
import hydra

@hydra.main(config_path=None)
def my_app(cfg):
    print(cfg.value, type(cfg.value))
    return 0

if __name__ == "__main__":
    my_app()
```

CLI 1:
`python examples/my_app.py hydra/sweeper=optuna hydra.sweeper.n_jobs=1 hydra.sweeper.n_trials=1 +value=null,1 hydra.sweeper.sampler.seed=2 -m`

CLI 2: (only `null` for value)
`python examples/my_app.py hydra/sweeper=optuna hydra.sweeper.n_jobs=1 hydra.sweeper.n_trials=1 +value=null hydra.sweeper.sampler.seed=2 -m`

**Stack trace/error message**

out for CLI 1:
```
...
None <class 'NoneType'>
...
```

out for CLI 2:
```
...
None <class 'str'>
...
```

## Expected Behavior

The type of `None` should always be `NoneType`. I.e.: `None <class 'NoneType'>` for both inputs.

## System information
- **Hydra Version** :  `1.1.1`
- **Python version** : `3.8.12`

## Additional context
Related to #1929 @Jasha10 ?

**Comment by Jasha10** [comment]
Hi @YannDubs,
The bug you've identified was fixed in #1818 and is not related to #1929.

Thanks for the nudge on this matter; I will release version 1.1.2 of the Optuna sweeper to pypi shortly.

**Comment by Jasha10** [comment]
Optuna sweeper v1.1.2 is now released:
https://pypi.org/project/hydra-optuna-sweeper/1.1.2
